types/mathjs/index.d.ts

```diff
@@ -1,2252 +1,4561 @@
-		gcd(...args: Fraction[]): Fraction ;
-		gcd(...args: MathArray[]): MathArray ;
-		gcd(...args: Matrix[]): Matrix;
-
-		/**
-		 * Calculate the hypotenusa of a list with values. The hypotenusa is defined as:
-		 * hypot(a, b, c, ...) = sqrt(a^2 + b^2 + c^2 + ...)
-		 * For matrix input, the hypotenusa is calculated for all values in the matrix.
-		 */
-		hypot(...args: number[]): number;
-		hypot(...args: BigNumber[]): BigNumber;
+declare namespace math {
+    // tslint:disable-line strict-export-declare-modifiers
+    type MathArray = number[] | number[][];
+    type MathType =
+        | number
+        | BigNumber
+        | Fraction
+        | Complex
+        | Unit
+        | MathArray
+        | Matrix;
+    type MathExpression = string | string[] | MathArray | Matrix;
+
+    interface MathJsStatic {
+        e: number;
+        pi: number;
+        i: number;
+        Infinity: number;
+        LN2: number;
+        LN10: number;
+        LOG2E: number;
+        LOG10E: number;
+        NaN: number;
+        null: number;
+        phi: number;
+        SQRT1_2: number;
+        SQRT2: number;
+        tau: number;
+
+        uninitialized: any;
+        version: string;
+
+        expression: MathNode;
+
+        /*************************************************************************
+         * Core functions
+         ************************************************************************/
+
+        /**
+         * Set configuration options for math.js, and get current options. Will
+         * emit a ‘config’ event, with arguments (curr, prev, changes).
+         * @param options Available options: {number} epsilon Minimum relative
+         * difference between two compared values, used by all comparison
+         * functions. {string} matrix A string ‘Matrix’ (default) or ‘Array’.
+         * {string} number A string ‘number’ (default), ‘BigNumber’, or
+         * ‘Fraction’ {number} precision The number of significant digits for
+         * BigNumbers. Not applicable for Numbers. {string} parenthesis How to
+         * display parentheses in LaTeX and string output. {string} randomSeed
+         * Random seed for seeded pseudo random number generator. Set to null to
+         * randomly seed.
+         * @returns Returns the current configuration
+         */
+        config: (options: object) => object;
+        /**
+         * Create a typed-function which checks the types of the arguments and
+         * can match them against multiple provided signatures. The
+         * typed-function automatically converts inputs in order to find a
+         * matching signature. Typed functions throw informative errors in case
+         * of wrong input arguments.
+         * @param name Optional name for the typed-function
+         * @param signatures Object with one or multiple function signatures
+         * @returns The created typed-function.
+         */
+        typed: (name: string, signatures: object) => any;
```

Comment:
This can be more strongly expressed:
```typescript
typed: (name: string, signatures: Record<string, (...args: any[]) => any>) => ((...args: any[]) => any);
```
or - even better - this is one of the rare places where `Function` is acceptable:
```typescript
typed: (name: string, signatures: Record<string, Function>) => Function; // tslint:disable-line no-Function <- or whatever that rule is, I can't remember off the top of my head.
```